File: 프로그래머스/2/42587. 프로세스/프로세스.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var queue = [(Int, Int)]()
    for (idx, prior) in priorities.enumerated() {
        queue.append((idx, prior))
    }
    var queueIdx = 0
    var cnt = 0
    
    while queueIdx < queue.count {
        let cur = queue[queueIdx]
        queueIdx += 1
        var temp = queueIdx
        var flag = false
        
        while temp < queue.count {
            if cur.1 < queue[temp].1 {
                flag = true
                break
            }
            temp += 1
        }
        if flag {
            queue.append(cur)
        } else {
            cnt += 1
            if cur.0 == location {
                return cnt
            }
        }
    }
    
    return cnt
}

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> priorities, int location) {
    queue<pair<int, int>> q;
    for(int i = 0; i < priorities.size(); i++) {
        q.push({priorities[i], i});
    }
    
    int cnt = 0;
    
    while(q.size()) {
        auto cur = q.front();
        q.pop();
        bool flag = false;
        queue<pair<int, int>> temp = q;
        while(temp.size()) {
            if (temp.front().first > cur.first) {
                flag = true;
                break;
            }
            temp.pop();
        }
        if(flag) {
            q.push(cur);
        } else {
            cnt++;
            if(cur.second == location) { return cnt; }
        }
    }
    
    return cnt;
}